net: http: http_client.c compilation fails with CONFIG_NET_SOCKETS_POSIX_NAMES=n (minor)

With Zephyr 3.3, having CONFIG_NET_SOCKETS_POSIX_NAMES **disabled**, compilation of `subsys/net/lib/http/http_client.c` fails with following error :

```
[262/322] Building C object zephyr/subsys/net/lib/http/CMakeFiles/subsys__net__lib__http.dir/http_client.c.obj
FAILED: zephyr/subsys/net/lib/http/CMakeFiles/subsys__net__lib__http.dir/http_client.c.obj 
ccache /home/lucas/zephyr-sdk-0.15.1/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DCORE_CM4 -DHSE_VALUE=8000000 -DKERNEL -DMBEDTLS_CONFIG_FILE=\"config-tls-generic.h\" -DSTM32F429xx -DUSE_FULL_LL_DRIVER -DUSE_HAL_DRIVER -D__PROGRAM_START -D__ZEPHYR__=1 -I/home/lucas/zephyrproject/zephyr/include -I/home/lucas/zephyrproject/http_client/build/zephyr/include/generated -I/home/lucas/zephyrproject/zephyr/soc/arm/st_stm32/stm32f4 -I/home/lucas/zephyrproject/zephyr/lib/posix/getopt/. -I/home/lucas/zephyrproject/zephyr/lib/util/fnmatch/. -I/home/lucas/zephyrproject/zephyr/drivers -I/home/lucas/zephyrproject/zephyr/soc/arm/st_stm32/common -I/home/lucas/zephyrproject/zephyr/subsys/net/l2 -I/home/lucas/zephyrproject/zephyr/subsys/net/lib/sockets/. -I/home/lucas/zephyrproject/zephyr/subsys/net/ip -I/home/lucas/zephyrproject/modules/hal/cmsis/CMSIS/Core/Include -I/home/lucas/zephyrproject/modules/hal/stm32/stm32cube/stm32f4xx/soc -I/home/lucas/zephyrproject/modules/hal/stm32/stm32cube/stm32f4xx/drivers/include -I/home/lucas/zephyrproject/modules/hal/stm32/stm32cube/stm32f4xx/drivers/include/Legacy -I/home/lucas/zephyrproject/modules/hal/stm32/stm32cube/common_ll/include -I/home/lucas/zephyrproject/modules/crypto/mbedtls/include -I/home/lucas/zephyrproject/modules/crypto/mbedtls/library -I/home/lucas/zephyrproject/zephyr/modules/mbedtls/configs -I/home/lucas/zephyrproject/zephyr/modules/mbedtls/include -isystem /home/lucas/zephyrproject/zephyr/lib/libc/minimal/include -isystem /home/lucas/zephyr-sdk-0.15.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/include -isystem /home/lucas/zephyr-sdk-0.15.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/include-fixed -fno-strict-aliasing -Os -imacros /home/lucas/zephyrproject/http_client/build/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m4 -mthumb -mabi=aapcs -mfp16-format=ieee --sysroot=/home/lucas/zephyr-sdk-0.15.1/arm-zephyr-eabi/arm-zephyr-eabi -imacros /home/lucas/zephyrproject/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/home/lucas/zephyrproject/http_client=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/lucas/zephyrproject/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/lucas/zephyrproject=WEST_TOPDIR -ffunction-sections -fdata-sections -D_POSIX_THREADS -std=c99 -nostdinc -MD -MT zephyr/subsys/net/lib/http/CMakeFiles/subsys__net__lib__http.dir/http_client.c.obj -MF zephyr/subsys/net/lib/http/CMakeFiles/subsys__net__lib__http.dir/http_client.c.obj.d -o zephyr/subsys/net/lib/http/CMakeFiles/subsys__net__lib__http.dir/http_client.c.obj -c /home/lucas/zephyrproject/zephyr/subsys/net/lib/http/http_client.c
/home/lucas/zephyrproject/zephyr/subsys/net/lib/http/http_client.c: In function 'http_wait_data':
/home/lucas/zephyrproject/zephyr/subsys/net/lib/http/http_client.c:427:25: error: 'POLLIN' undeclared (first use in this function)
  427 |         fds[0].events = POLLIN;
      |                         ^~~~~~
/home/lucas/zephyrproject/zephyr/subsys/net/lib/http/http_client.c:427:25: note: each undeclared identifier is reported only once for each function it appears in
```

**To Reproduce**
Steps to reproduce the behavior:
1. Consider sample `samples/net/sockets/http_client`
2. Disable `CONFIG_NET_SOCKETS_POSIX_NAMES` in `prj.conf`
3. Build

Build will fail and generate the error above.

**Expected behavior**
`http_client.c` should compile without error with CONFIG_NET_SOCKETS_POSIX_NAMES disabled.

**Impact**
Cannot disable CONFIG_NET_SOCKETS_POSIX_NAMES

**Environment (please complete the following information):**
 - Linux
 - `zephyr-sdk-0.15.1`
 - Zephyr 3.3.0

Solutions would be either to include `include/zephyr/posix/poll.h` at the beginning of `subsys/net/lib/http/http_client.c` or replace `POLLIN` by `ZSOCK_POLLIN` line 427: https://github.com/zephyrproject-rtos/zephyr/blob/07c6af3b8c35c1e49186578ca61a25c76e2fb308/subsys/net/lib/http/http_client.c#L427

--- a/subsys/net/lib/http/http_client.c
+++ b/subsys/net/lib/http/http_client.c
@@ -424,7 +424,7 @@ static int http_wait_data(int sock, struct http_request *req, int32_t timeout)
 	int64_t timestamp = k_uptime_get();
 
 	fds[0].fd = sock;
-	fds[0].events = POLLIN;
+	fds[0].events = ZSOCK_POLLIN;
 
 	do {
 		if (timeout > 0) {

